Makefile

```diff
@@ -1,263 +1,165 @@
+	@echo  '  themes.*    - bootstrap CSS / oscar / simple'
+	@echo  '  docker      - build Docker image'
+	@echo  '  gecko.driver'
+	@echo  '  - download & install geckodriver if not already installed'
+	@echo  '  search.checker.<engine>'
+	@echo  '  - check search results of <enigne>'
 	@$(MAKE) -e -s make-help
 
-help-all: help-min
-	@echo  ''
-	@$(MAKE) -e -s python-help
-	@echo  ''
-	@$(MAKE) -e -s docs-help
-
-PHONY += install
-install: buildenv pyenvinstall
-
-PHONY += uninstall
-uninstall: pyenvuninstall
-
-PHONY += clean
-clean: pyclean docs-clean node.clean test.clean
-	$(call cmd,common_clean)
-
 PHONY += run
-run:  buildenv pyenvinstall
+run:  install
 	$(Q) ( \
 	sleep 2 ; \
 	xdg-open http://127.0.0.1:8888/ ; \
 	) &
-	SEARX_DEBUG=1 $(PY_ENV)/bin/python ./searx/webapp.py
+	SEARX_DEBUG=1 $(MTOOLS) pyenv.cmd python ./searx/webapp.py
+
+PHONY += install uninstall
+install:
+	$(Q)$(MTOOLS) pyenv.OK >/dev/null || $(MTOOLS) build_msg PYENV  "not OK, needs rebuild .."
+	$(Q)$(MTOOLS) pyenv.assert && $(MTOOLS) build_msg PYENV  "OK"
+
+uninstall:
+	$(Q)$(MTOOLS) pyenv.uninstall
+
+PHONY += clean
+clean: pyclean docs.clean node.clean test.clean
+	$(Q)$(MTOOLS) build_msg CLEAN  "common files"
+	$(Q)find . -name '*.orig' -exec rm -f {} +
+	$(Q)find . -name '*.rej' -exec rm -f {} +
+	$(Q)find . -name '*~' -exec rm -f {} +
+	$(Q)find . -name '*.bak' -exec rm -f {} +
+
+PHONY += pyclean pybuild upload-pypi upload-pypi-test
+pyclean pybuild:
+	$(Q)$(MTOOLS) $@
+
+upload-pypi:
+	$(Q)$(MTOOLS) pypi.upload
+
+upload-pypi-test:
+	$(Q)$(MTOOLS) pypi.upload.test
 
 # docs
 # ----
 
-sphinx-doc-prebuilds:: buildenv pyenvinstall prebuild-includes
+PHONY += docs docs.clean sphinx-doc-prebuilds docs.live
+
+sphinx-doc-prebuilds:: install buildenv docs.prebuild
 
-PHONY += docs
 docs:  sphinx-doc-prebuilds
-	$(call cmd,sphinx,html,docs,docs)
+	$(Q)$(MTOOLS) docs.html
 
-PHONY += docs-live
-docs-live:  sphinx-doc-prebuilds
-	$(call cmd,sphinx_autobuild,html,docs,docs)
+docs.live: sphinx-doc-prebuilds
+	$(Q)$(MTOOLS) docs.autobuild
 
-PHONY += prebuild-includes
-prebuild-includes:
-	$(Q)mkdir -p $(DOCS_BUILD)/includes
-	$(Q)./utils/searx.sh doc | cat > $(DOCS_BUILD)/includes/searx.rst
-	$(Q)./utils/filtron.sh doc | cat > $(DOCS_BUILD)/includes/filtron.rst
-	$(Q)./utils/morty.sh doc | cat > $(DOCS_BUILD)/includes/morty.rst
+docs.prebuild docs.clean docs.html:
+	$(Q)$(MTOOLS) $@
 
+PHONY += docs-clean docs-live
+docs-clean: docs.clean
+	$(Q)$(MTOOLS) build_msg WARN $@ is deprecated use docs.clean
 
-$(GH_PAGES)::
-	@echo "doc available at --> $(DOCS_URL)"
+docs-live: docs.live
+	$(Q)$(MTOOLS) build_msg WARN $@ is deprecated use docs.live
 
 # update project files
 # --------------------
 
-PHONY += project engines.languages useragents.update buildenv
+PHONY += ci.data ci.themes ci.gh-pages engines.languages useragents.update buildenv
+ci.data ci.themes ci.gh-pages engines.languages useragents.update buildenv::
```

Comment:
> These targets are not exclusively design for the CI.

Even this targets are named ci, they are not dedicated only to ci .. ci. here means _compile all themes_  

But it might be better to name them themes.all ... and so on .. to be more precisely.

> Also to rename: ... to data.xyz

make  sense to me. I will do it.

Reply:
fixed: have a look at the new [name schema](https://return42.github.io/searx/dev/makefile.html) , there are only some [make-targets left](https://github.com/return42/searx/blob/manage-script/Makefile#L16-L23)